cashctrl_ledger/ledger.py

```diff
@@ -47,11 +205,128 @@ def add_price():
-        """
-        Not implemented yet
+    def add_vat_code(
+        self, code: str, rate: float, account: str, inclusive: bool = True,
+        text: str = ""
+    ):
         """
-        raise NotImplementedError
+        Adds a new VAT code to the CashCtrl account.
+
+        Parameters:
+            code (str): The VAT code to be added.
+            rate (float): The percentage rate of the VAT.
+            account (str): The account identifier to which the VAT is applied.
+            inclusive (bool): Determines whether the VAT is calculated as 'NET' 
+                            (True, default) or 'GROSS' (False).
+            text (str): Additional text or description associated with the VAT code.
+
+        Returns:
+            None: This method does not return any value but raises exceptions if
+                the input parameters are not valid or the API request fails.
+
+        Raises:
+            ValueError: If any of the inputs are invalid (wrong type or out of allowed range).
+            Exception: If there is an issue with the server connection or the API request.
+        """
+        # Prepare payload for the API request
+        payload = {
+            "name": code,
+            "percentage": rate,
+            "accountId": account,
+            "calcType": "NET" if inclusive else "GROSS",
+            "documentName": text,
+        }
+
+        if not isinstance(payload['name'], str) or len(payload['name']) > 50:
+            raise ValueError(
+                "Invalid name. It must be a string with a maximum of 50 characters."
+            )
+
+        if not isinstance(payload['percentage'], (int, float)) \
+                or not (0.0 <= payload['percentage'] <= 100.0):
+            raise ValueError(
+                "Invalid percentage. It must be a number between 0.0 and 100.0."
+            )
+
+        if payload['calcType'] not in ["NET", "GROSS"]:
+            raise ValueError("Invalid calcType. It must be either 'NET' or 'GROSS'.")
+
+        if not isinstance(payload['documentName'], str) \
+                or len(payload['documentName']) > 50:
+            raise ValueError(
+                "Invalid documentName. It must be a string with a maximum of 50 characters."
+            )
+
+        try:
+            self.client.post("tax/create.json", data=payload)
+        except Exception as e:
+            raise Exception(
+                f"An error occurred while posting data to the server: {e}"
+            )
+        
+    def update_vat_code(
+        self, code: str, rate: float, account: str,
+        inclusive: bool = True, text: str = "", name: str = "",
+    ):
+        """
+        Updates an existing VAT code in the CashCtrl account with new parameters. 
+        If a VAT code is already in use, some parameters like 'percentage' might not be changeable.
+
+        Parameters:
+            code (str): The unique identifier for the VAT code to be updated, max 50 characters.
```

Comment:
`code` is the name of the vat_code entry to modify. In pyledger, we do not know anything about CashCtrl's internal ids.